src/adios4dolfinx/adios2_helpers.py

```diff
@@ -185,11 +185,17 @@ def read_array(
 
     arr = io.InquireVariable(array_name)
     arr_shape = arr.Shape()
-    assert len(arr_shape) == 1
+    assert len(arr_shape) >= 1  # TODO: Should we always pick the first element?
     arr_range = compute_local_range(comm, arr_shape[0])
-    assert arr_range[0] == arr_range[0]
-    arr.SetSelection([[arr_range[0]], [arr_range[1] - arr_range[0]]])
-    vals = np.empty(arr_range[1] - arr_range[0], dtype=np.dtype(arr.Type().strip("_t")))
+    assert arr_range[0] == arr_range[0]  # FIXME: Should this be arr_shape?
```

Comment:
This doesn't test anything?:P always true?:)

Reply:
I agree, but the line was there from before I I assumed maybe you wanted to test
```
arr_range[0] == arr_shape[0]
```
or something similar?

Reply:
Probably